lib/search/best_bets_checker.rb

```diff
@@ -9,16 +9,12 @@ def initialize(query, index = nil)
     end
 
     def best_bets
-      unless @fetched
-        fetch
-      end
+      @fetched || fetch
```

Comment:
Since you're not using the value, I think this is slightly clearer with `unless`, but on a single line: `fetch unless @fetched`

The confusing part to me is the attributes are being set by side effects of `fetch`, so the behaviour depends on previous calls.

You could perhaps avoid this by splitting the class up, but up to you whether that's worth doing. I'm thinking of something like
1. a value object for the combined best/worst bets
2. a thing that takes a query, fetches the data and does the combining